src/components/pick-pane-panel/pick-pane-panel.tsx

```diff
@@ -0,0 +1,187 @@
+  margin: 0;
+  background-image: ${({ image }) => `url(${image})`};
+  background-size: auto;
+  background-repeat: no-repeat;
+  cursor: pointer;
+  border-width: 0;
+  border-radius: 8px;
+
+  ${({ active = false }) =>
+    active &&
+    css`
+      box-shadow: 0 0 0 2px #000000, 0 0 0 4px #ffffff;
+      // -moz-box-shadow: 0 0 0 2px #000000, 0 0 0 4px #ffffff;
+      // -webkit-shadow: 0 0 0 2px #000000, 0 0 0 4px #ffffff;
+    `}
+`;
+
+interface PickPanePanelOptions {
+  panelLength: number;
+  selectedKey: string;
+}
+
+type PickPanePanelProps = {
+  pickPaneSetName: PickPaneSetName;
+  group: string;
+  panelLength: number;
+  imageWidth: number;
+  imageHeight: number;
+};
+
+export const PickPanePanel = ({
+  pickPaneSetName,
+  group,
+  panelLength,
+  imageWidth,
+  imageHeight,
+}: PickPanePanelProps) => {
+  const dispatch = useAppDispatch();
+
+  const imageArray = useAppSelector(selectPickPaneArray(pickPaneSetName));
+  const imagePickedKey = useAppSelector(
+    selectPickPanePickedId(pickPaneSetName)
+  );
+
+  const renderTextureArray = (
+    array: IPickPane[],
+    options: PickPanePanelOptions
+  ): JSX.Element[] => {
+    let elements: JSX.Element[] = [];
+    for (let i = 0; i < array.length; i += options.panelLength) {
+      elements = elements.concat(renderTexturePanel(array, i, options));
+    }
+    return elements;
+  };
+
+  const renderTexturePanel = (
+    array: IPickPane[],
+    start: number,
+    options: PickPanePanelOptions
+  ): JSX.Element => {
+    const elements: JSX.Element[] = [];
+
+    const startIndex = start >= array.length ? array.length - 1 : start;
+    const endIndex =
+      startIndex + options.panelLength >= array.length
+        ? array.length
+        : startIndex + options.panelLength;
+
+    for (let i = startIndex; i < endIndex; i++) {
+      const item = array[i];
+      if (item.group === group) {
+        elements.push(
+          <TextureImageWrapper
+            imageName={item.name || ""}
+            active={!!item.name && imagePickedKey === item.id}
+          >
+            <TextureImage
+              key={`${item.id}`}
+              image={`${item.icon}`}
+              width={imageWidth}
+              height={imageHeight}
+              active={!item.name && imagePickedKey === item.id}
+              onClick={() => {
+                dispatch(
+                  setPickPanePicked({
+                    pickPaneSetName: pickPaneSetName,
+                    id: item.id,
+                  })
+                );
+              }}
+            ></TextureImage>
+            <TextureImageName imageName={item.name || ""}>
+              {item.name || ""}
+            </TextureImageName>
+          </TextureImageWrapper>
+        );
+      }
+    }
+
+    return <TexturePanel>{elements}</TexturePanel>;
+  };
+
+  return (
+    <>
+      {renderTextureArray(imageArray, {
+        panelLength: panelLength,
+        selectedKey: imagePickedKey,
+      })}
```

Comment:
Write JSX here. No sense to call just one function.

Reply:
Done.